inlong-manager/manager-client-examples/src/test/java/org/apache/inlong/manager/client/File2IcebergExample.java

```diff
@@ -0,0 +1,160 @@
+        configuration.setAuthentication(super.getInlongAuth());
+        InlongClient inlongClient = InlongClient.create(super.getServiceUrl(), configuration);
+        InlongGroupInfo groupInfo = createGroupInfo();
+        try {
+            InlongGroup group = inlongClient.forGroup(groupInfo);
+            InlongGroupContext groupContext = group.delete();
+            Assert.notNull(groupContext);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private FileSource createAgentFileSource() {
+        FileSource fileSource = new FileSource();
+        fileSource.setSourceName("{source.name}");
+        fileSource.setAgentIp("{agent.ip}");
+        fileSource.setPattern("/a/b/*.txt");
+        fileSource.setTimeOffset("-1h");
+        return fileSource;
+    }
+
+    private List<StreamField> createStreamFields() {
+        List<StreamField> streamFieldList = Lists.newArrayList();
+        streamFieldList.add(new StreamField(0, FieldType.STRING.toString(), "name", null, null));
+        streamFieldList.add(new StreamField(1, FieldType.INT.toString(), "age", null, null));
+        streamFieldList.add(new StreamField(2, FieldType.DECIMAL.toString(), "score", null, null));
+        streamFieldList.add(new StreamField(3, FieldType.TIMESTAMP.toString(), "ts", null, null));
+        return streamFieldList;
+    }
+
+    /**
+     * Create iceberg sink
+     */
+    public IcebergSink createIcebergSink() throws Exception {
+        IcebergSink sink = new IcebergSink();
+
+        sink.setSinkName("{sink.name}");
+        sink.setDbName("{db.name}");
+        sink.setTableName("{table.name}");
+        sink.setCatalogUri("thrift://{ip:port}");
+        sink.setWarehouse("hdfs://{ip:port}/user/iceberg/warehouse/");
+
+        final SinkField field1 = new SinkField(0, FieldType.INT.toString(), "age", FieldType.INT.toString(), "age");
+        final SinkField field2 = new SinkField(1, FieldType.STRING.toString(), "name", FieldType.STRING.toString(),
+                "name");
+        final SinkField field3 = new SinkField(3, FieldType.DECIMAL.toString(), "score", FieldType.DECIMAL.toString(),
+                "score");
+        final SinkField field4 = new SinkField(3, FieldType.TIMESTAMP.toString(), "ts", FieldType.TIMESTAMP.toString(),
+                "ts");
+
+        // field ext param
+        // field1: bucket partition example
+        final ObjectMapper mapper = new ObjectMapper();
+        IcebergColumnInfo info1 = new IcebergColumnInfo();
+        info1.setRequired(true);
+        info1.setPartitionStrategy(IcebergPartition.BUCKET.toString());
+        info1.setBucketNum(10);
+        field1.setExtParams(mapper.writeValueAsString(info1));
+
+        // field3: decimal column example
+        IcebergColumnInfo info3 = new IcebergColumnInfo();
+        info3.setScale(5);
+        info3.setPrecision(10);  //NOTE: scale must be less than or equal to precision
```

Comment:
Did this restriction checked in save or update APIs?

Reply:
Not yet. The server end does not look into field extended param string.  The exception is thrown when creating sink resources, aka iceberg tables.

Reply:
Added validation on field save or update.